workspace-manager-clienttests/src/main/java/scripts/testscripts/CreateGetDeleteControlledGcsBucket.java

```diff
@@ -0,0 +1,211 @@
+    assertThat(bucket.getErrorReport().getStatusCode(), equalTo(HttpStatusCodes.STATUS_CODE_BAD_REQUEST));
+
+    // Create the cloud context
+    logger.info("Creating cloud context");
+
+    String contextJobId = UUID.randomUUID().toString();
+    var createContext = new CreateCloudContextRequest()
+    .cloudPlatform(CloudPlatform.GCP)
+    .jobControl(new JobControl().id(contextJobId));
+    CreateCloudContextResult contextResult = workspaceApi.createCloudContext(createContext, getWorkspaceId());
+    while (ClientTestUtils.jobIsRunning(contextResult.getJobReport())) {
+      TimeUnit.SECONDS.sleep(CREATE_CONTEXT_POLL_SECONDS);
+      contextResult = workspaceApi.getCreateCloudContextResult(getWorkspaceId(), contextJobId);
+    }
+    logger.info("Create context status is {}", contextResult.getJobReport().getStatus().toString());
+    assertThat(contextResult.getJobReport().getStatus(), equalTo(JobReport.StatusEnum.SUCCEEDED));
+
+    // Create the bucket - should work this time
+    bucket = createBucketAttempt(resourceApi);
+    assertThat(bucket.getJobReport().getStatus(), equalTo(JobReport.StatusEnum.SUCCEEDED));
+    UUID resourceId = bucket.getResourceId();
+
+    // Retrieve the bucket resource
+    logger.info("Retrieving bucket resource id {}", resourceId.toString());
+    GcsBucketStoredAttributes gotBucket = resourceApi.getBucket(getWorkspaceId(), resourceId);
+    assertThat(gotBucket.getBucketName(), equalTo(bucket.getGcpBucket().getBucketName()));
+
+    // TODO: Check access:
+    // - writer can add the file
+    // - writer can read the file
+    // - reader can read the file
+    // - reader cannot write a file
+    // - reader cannot delete the bucket
+
+    // Delete bucket
+    String deleteJobId = UUID.randomUUID().toString();
+    var deleteRequest =
+        new DeleteControlledGcsBucketRequest().jobControl(new JobControl().id(deleteJobId));
+    logger.info("Deleting bucket resource id {} jobId {}", resourceId, deleteJobId);
+    DeleteControlledGcsBucketResult result =
+        resourceApi.deleteBucket(deleteRequest, getWorkspaceId(), resourceId);
+    while (ClientTestUtils.jobIsRunning(result.getJobReport())) {
+      TimeUnit.SECONDS.sleep(DELETE_BUCKET_POLL_SECONDS);
+      result = resourceApi.getDeleteBucketResult(getWorkspaceId(), deleteJobId);
+    }
+    logger.info("Delete bucket status is {}", result.getJobReport().getStatus().toString());
+    assertThat(result.getJobReport().getStatus(), equalTo(JobReport.StatusEnum.SUCCEEDED));
+
+    // verify it's not there anymore
+    // - via metadata
+    // TODO: via GCP access
+
+    try {
+      resourceApi.getBucket(getWorkspaceId(), resourceId);
+      throw new IllegalStateException("Incorrectly found a deleted bucket!");
+    } catch (ApiException ex) {
+      assertThat(ex.getCode(), equalTo(HttpStatusCodes.STATUS_CODE_NOT_FOUND));
+    }
+    bucketName = null;
+
+    // Delete the cloud context
```

Comment:
nit: this isn't required, the fixture will delete the workpace and that flight includes deleting the GCP project.

Reply:
Added a comment. Seems worth exercising deleting the cloud context.